### PR TITLE
Adding components.template file

### PR DESCRIPTION
**Pull request by ollips**
Link: preview-543

Adding `components.template` file which is a good starting point when writing a doc page for a new component. Similar to the `atomics.template`.

### CODE REVIEW

File: site/src/components.template [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  - <component-name>
+---
+
+# Page title (component name)
+
+The first thing is the general description.
+
+Consider using tables to organize information if there are a lot of nuances.
+
+| cssproperty | value                           | screensize |
```
Comment (wibjorn) [inline site/src/components.template:17]: This is for atomics, isn't it?

File: site/src/components.template [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+```html
+<div>
+  <button class="button">Click me</button>
+</div>
+```
+
+### Modifier heading
+
+If there are components variations available such as different semantic colors on buttons, different hero sizes, loading states, etc., the next section(s) should be about them.
```
Comment (homing1) [inline site/src/components.template:37]:
```suggestion
If there are components variations available such as different semantic colors, sizes, loading states, etc., the next section(s) should be about them.
```

File: site/src/components.template (hunk 1 above)
Comment (ollips) [reply site/src/components.template:17]: Tables can still be used in components docs, so I thought it worth mentioning

- https://design.learn.microsoft.com/components/button.html
- https://design.learn.microsoft.com/components/checkbox.html
- https://design.learn.microsoft.com/components/gradients.html
- https://design.learn.microsoft.com/components/help.html

File: site/src/components.template (hunk 2 above)
Comment (ollips) [reply site/src/components.template]: Updated, thanks!

File: site/src/components.template (hunk 1 above)
Comment (wibjorn) [reply site/src/components.template:17]: They can, but they won't have the headings `cssproperty | value | screensize`. I would suggest we remove it or at the very least update the headings.

Comment (ollips) [reply site/src/components.template]: Good point, kept the one with ` State | Class | Example ` titles

**Review by wibjorn — APPROVED** [review]

**Review by homing1 — APPROVED** [review]

**Pull request merged**